vela-templates/definitions/internal/trait/gateway.cue

```diff
@@ -14,10 +14,20 @@ gateway: {
 				}
 				if len(igs) > 0 {
 				  if igs[0].ip != _|_ {
-					  message: "Visiting URL: " + context.outputs.ingress.spec.rules[0].host + ", IP: " + igs[0].ip
+				  	if igs[0].host != _|_ {
+					    message: "Visiting URL: " + context.outputs.ingress.spec.rules[0].host + ", IP: " + igs[0].ip
+				  	}
+				  	if igs[0].host == _|_ {
+					    message: "Visiting URL: " + "localhost" + ", IP: " + igs[0].ip
+				  	}
 				  }
 				  if igs[0].ip == _|_ {
-					  message: "Visiting URL: " + context.outputs.ingress.spec.rules[0].host
+				  	if igs[0].host != _|_ {
+						  message: "Visiting URL: " + context.outputs.ingress.spec.rules[0].host
+						}
+				  	if igs[0].host != _|_ {
+				  		message: "Visiting URL: " + "localhost"
```

Comment:
the localhost address maybe is misleading

Reply:
Fixed, "access localhost" infomation moved to velad as a spectial situation